### ISSUE TITLE
Why some audio greedy decoding outputs are all <oov>？

### ISSUE THREAD

**Issue by ziyu123** [k2-fsa/icefall#1209]
I am training an english zipformer2 model, but some audio decode are all <oox>, like this 
000086	 ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇ ' ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇ ' ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇   ⁇ 
When I concat the above audio together, it will output normally  and When I trim the previous few seconds of audio, some will output normally and some will not.  
What is the cause of this?

**Comment by ziyu123** [comment]
I force to set logits[:,2] += -999999.0 when decoding.

**Comment by ziyu123** [comment]
I force to set logits[:,2] += -999999.0 when decoding.

**Comment by ziyu123** [comment]
I force to set logits[:,2] += -999999.0 when decoding.

**Comment by csukuangfj** [comment]
> I force to set logits[:,2] += -999999.0 when decoding.

Does it fix your issue?

**Comment by csukuangfj** [comment]
If yes, you can modify the code to treat `unk_id` as `blank_id`.

**Comment by ziyu123** [comment]
> If yes, you can modify the code to treat `unk_id` as `blank_id`.

thanks a lot,  set logits[:,2] solved my problem, if treat `unk_id` as `blank_id`, I think it's also ok.

**Comment by csukuangfj** [comment]
We are treating unk_id as blank_id:
https://github.com/k2-fsa/icefall/blob/00256a766921dd34a267012b0e2b8ff7d538f0e6/egs/librispeech/ASR/pruned_transducer_stateless2/beam_search.py#L631

https://github.com/k2-fsa/icefall/blob/00256a766921dd34a267012b0e2b8ff7d538f0e6/egs/librispeech/ASR/pruned_transducer_stateless2/beam_search.py#L746

Which decoding method and which file are you using?

**Comment by danpovey** [comment]
Maybe some training utts were all unk due to some encoding issue or
language mismatch?

**Comment by ziyu123** [comment]
> Maybe some training utts were all unk due to some encoding issue or language mismatch?

i‘m very sorry,  IT'S MY TRAIN DATA ERROR, some utts in my train text are lowercase, The lowercase English letters cause the sentence to be all  <unk>
